src/spec/expectations.cr

```diff
@@ -417,6 +417,26 @@ module Spec
   end
 
   module ObjectExtensions
+    # Validates an expectation and fails the example if it does not match.
+    #
+    # This overload will be called when doing:
+    #
+    # ```
+    # result = subject.should be_a(Type)
+    # # result will be a Type
+    # ```
+    #
+    # and make sure, at compile-time, that the returned object is of type `Type`.
+    #
+    # See `Spec::Expecations` for available expectations.
+    def should(expectation : BeAExpectation(T), file = __FILE__, line = __LINE__) forall T
```

Comment:
Maybe
```suggestion
    def should(expectation : BeAExpectation(T), file = __FILE__, line = __LINE__) : T forall T
```

Reply:
Done!

Reply:
I think you added it to the wrong method, the type restriction only works on `should`.

Reply:
Other thing: What do you think about adding `[... whose type is restricted] to the expected type.`?

Reply:
Where?

Reply:
Damn, I commented on the wrong line. I'll make a new suggestion.